src/pages/index.tsx

```diff
@@ -83,59 +83,75 @@ export const defaultCollections = (
     );
   });
 
-export default function HomePage() {
-  const router = useRouter();
+const HomePage: NextPage<{
+  landingPage: LandingPageQuery['landingPage'];
+  page: PageQuery['page'];
+}> = ({ landingPage, page }) => {
   const locale = useLocale();
   const {
     utils: { getRoutedInternalHref },
   } = useConfig();
-  const language = getQlLanguage(
-    router.locale ?? router.defaultLocale ?? DEFAULT_LANGUAGE
-  );
-  const { data } = useQuery(LANDING_PAGE_QUERY, {
-    variables: {
-      languageCode: language,
-    },
-  });
   return (
-    <RHHCApolloPage
-      uri={getI18nPath('/', locale)}
+    <HCRCPage
       className="pageLayout"
       navigation={<Navigation />}
       content={
-        <RHHCApolloPageContent
+        <HCRCPageContent
           breadcrumbs={[]}
-          landingPage={data.landingPage}
+          page={page}
+          landingPage={landingPage}
           PageContentLayoutComponent={LandingPageContentLayout}
-          collections={(page) =>
+          collections={(page: PageContentProps['page']) =>
             defaultCollections(page, getRoutedInternalHref)
           }
+          language={getQlLanguage(locale)}
         />
       }
       footer={<FooterSection />}
     />
   );
-}
+};
 
 export async function getStaticProps(context: GetStaticPropsContext) {
   return getHobbiesStaticProps(context, async ({ cmsClient }) => {
     const locale = context.locale ?? context.defaultLocale ?? DEFAULT_LANGUAGE;
-    // TODO: Stop using landing page here when the front page can be queried from the Headless CMS.
-    // The front page should be queried instead - there is no need for landing page.
-    await cmsClient.query({
-      query: LANDING_PAGE_QUERY,
+    const { data: landingPageData } = await cmsClient.query<
+      LandingPageQuery,
+      LandingPageQueryVariables
+    >({
+      query: LandingPageDocument,
       variables: {
+        id: 'root',
         languageCode: getQlLanguage(locale),
       },
     });
 
+    const { data: pageData } = await cmsClient.query<
+      PageByTemplateQuery,
+      PageByTemplateQueryVariables
+    >({
+      query: PageByTemplateDocument,
+      variables: {
+        template: TemplateEnum.FrontPage,
+        language: getQlLanguage(locale).toLocaleLowerCase(),
+      },
+    });
+
+    const page = pageData.pageByTemplate;
+
+    const landingPage = landingPageData.landingPage;
+
     return {
       props: {
         ...(await serverSideTranslationsWithCommon(getLocaleOrError(locale), [
           'home',
           'search',
         ])),
+        landingPage: landingPage,
+        page: page,
```

Comment:
So, do we still need both, the front page and the landing page? They have overlapping fields and features which makes the use of them complicated and hard together. It's a good thing that both the queries are implemented in HCRC-lib and it should be decision of the application to make which query it uses, but using them both together is not a good thing. If both are still needed, then we need changes from the headless cms -team.